var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -805,9 +805,22 @@ def find_optional_library(name, prefix):
                 "apf_zoltan",
                 "spr",
             ]
+            pumi_dep_zoltan = [""]
+            pumi_dep_parmetis = [""]
+            if "+zoltan" in spec["pumi"]:
+                pumi_dep_zoltan = ld_flags_from_dirs([spec["zoltan"].prefix.lib], ["zoltan"])
+                if "+parmetis" in spec["zoltan"]:
+                    pumi_dep_parmetis = ld_flags_from_dirs(
+                        [spec["parmetis"].prefix.lib], ["parmetis"]
+                    )
             options += [
                 "PUMI_OPT=-I%s" % spec["pumi"].prefix.include,
-                "PUMI_LIB=%s" % ld_flags_from_dirs([spec["pumi"].prefix.lib], pumi_libs),
+                "PUMI_LIB=%s %s %s"
+                % (
+                    ld_flags_from_dirs([spec["pumi"].prefix.lib], pumi_libs),
+                    pumi_dep_parmetis,
+                    pumi_dep_zoltan,
```

Comment:
If I understand correctly, Zoltan depends on ParMETIS, so it is probably better to move `pumi_dep_parmetis` after `pumi_dep_zoltan`.

Reply:
I'm not if this is always a requirement but at least in some cases if `A` depends on `B` one needs to put `-lA` before `-lB`  -- this was my reasoning for the suggestion.

Reply:
I just pushed the change https://github.com/spack/spack/pull/34864/commits/5084b1bef49b818046742249179b060fe6f66138.